modules/charts/src/components/Chart.tsx

```diff
@@ -23,10 +23,23 @@ type ChartProps = {
 	DisplayComponent?: React.ReactElement<ArrangerChartProps>;
 };
 
-// TODO: numeric or agg, very hacky property check, no need for this once server config is expanded, then we can check the typename
-const resolveData = ({ data }) => {
-	if (data.buckets) return data.buckets;
-	if (data.range) return data.range.buckets;
+// TODO: numeric or agg, very hacky property check,
+//  no need for this once server config is expanded, then we can check the typename
+const resolveData = ({ data, onDataLoad }) => {
+	let chartData = {};
+
+	if (data.buckets) {
+		chartData = data.buckets;
+	} else if (data.range) {
+		chartData = data.range.buckets;
+	}
+
+	// consumer
+	if (onDataLoad) {
+		chartData = onDataLoad(chartData);
+	}
+
+	return chartData;
 };
 
 /**
```

Comment:
Arranger Charts currently only supports "aggregations" field of Arranger Server
There's nothing implemented to grab a total for field, this allows the client to at least grab all the buckets and compute a total. escape hatch until Arranger Charts is feature complete.

Reply:
additionally we will need to send back the typenames for the fields,